File: 황영훈/week11/성격_유형_검사하기.java
class Solution {
    public String solution(String[] survey, int[] choices) {
        // 변수 초기화
        String answer = "";
        int[] scores = new int[4];
        String[] leftTypes = {"R", "C", "J", "A"};
        String[] rightTypes = {"T", "F", "M", "N"};
        
        // 문항마다 성격유형과 점수를 계산
        for (int i = 0; i < choices.length; i++) {
            String type = "";
            int score = 0;
            if (choices[i] < 4) {
                type = survey[i].substring(0,1);
                score = 4 - choices[i];
            }
            if (choices[i] > 4) {
                type = survey[i].substring(1);
                score = choices[i] - 4;
            }
            for (int j = 0; j < 4; j++) {
                if (leftTypes[j].equals(type)) {
                    scores[j] += score;
                    break;
                }
                if (rightTypes[j].equals(type)) {
                    scores[j] -= score;
                    break;
                }
            }
        }
        
        // 더 높은 점수를 가진 유형을 선택
        for (int i = 0; i < 4; i++) {
            answer += (scores[i] >= 0) ? leftTypes[i] : rightTypes[i];
        }
        
        return answer;
    }
}

File: 이수빈/week11/성격유형검사하기.py
def solution(survey, choices):
    answer = ''
    score = {'R': 0, 'T': 0, 'C': 0, 'F': 0, 'J': 0, 'M': 0, 'A': 0, 'N': 0} # 점수 담는 dictionary
    
    for i in range(len(survey)):
        disagree_type = survey[i][0] # 비동의 관련 선택지를 선택하면 받는 성격 유형
        agree_type = survey[i][1] # 동의 관련 선택지를 선택하면 받는 성격 유형
        choice = choices[i]
        
        if choice == 1:
            score[disagree_type] += 3
        elif choice == 2:
            score[disagree_type] += 2
        elif choice == 3:
            score[disagree_type] += 1
        elif choice == 5:
            score[agree_type] += 1
        elif choice == 6:
            score[agree_type] += 2
        elif choice == 7:
            score[agree_type] += 3
    
    if score['R'] < score['T']:
        answer += 'T'
    else:
        answer += 'R'
    if score['C'] < score['F']:
        answer += 'F'
    else:
        answer += 'C'
    if score['J'] < score['M']:
        answer += 'M'
    else:
        answer += 'J'
    if score['A'] < score['N']:
        answer += 'N'
    else:
        answer += 'A'
            
    return answer